src/main/resources/db/migration/v0.21.0.1__add_allele_database_status.sql

```diff
@@ -0,0 +1,30 @@
+CREATE TABLE alleledatabasestatusslotannotation (
+	id bigint PRIMARY KEY,
+	singleallele_curie varchar(255),
+	databasestatus_id bigint
+);
+
+ALTER TABLE alleledatabasestatusslotannotation ADD CONSTRAINT alleledatabasestatus_singleallele_curie_fk FOREIGN KEY (singleallele_curie) REFERENCES allele (curie);
+ALTER TABLE alleledatabasestatusslotannotation ADD CONSTRAINT alleledatabasestatus_databasestatus_id_fk FOREIGN KEY (databasestatus_id) REFERENCES vocabularyterm (id);
+
+CREATE INDEX alleledatabasestatus_singleallele_curie_index ON alleledatabasestatusslotannotation USING btree (singleallele_curie);
+CREATE INDEX alleledatabasestatus_databasestatus_id_index ON alleledatabasestatusslotannotation USING btree (databasestatus_id);
+
+CREATE TABLE alleledatabasestatusslotannotation_aud (
+    id bigint NOT NULL,
+    rev integer NOT NULL,
+	singleallele_curie varchar(255),
+	databasestatus_id bigint
+);
+
+ALTER TABLE alleledatabasestatusslotannotation_aud ADD PRIMARY KEY (id, rev);
+
+ALTER TABLE alleledatabasestatusslotannotation_aud ADD CONSTRAINT alleledatabasestatus_aud_id_rev_fk FOREIGN KEY (id, rev) REFERENCES slotannotation_aud (id, rev);
+
+INSERT INTO vocabulary (id, name) VALUES (nextval('hibernate_sequence'), 'Allele database status vocabulary');
+	
+INSERT INTO vocabularyterm (id, name, vocabulary_id) SELECT nextval('hibernate_sequence'), 'approved', id FROM vocabulary WHERE name = 'Allele database status vocabulary';
+INSERT INTO vocabularyterm (id, name, vocabulary_id) SELECT nextval('hibernate_sequence'), 'deleted', id FROM vocabulary WHERE name = 'Allele database status vocabulary';
+INSERT INTO vocabularyterm (id, name, vocabulary_id) SELECT nextval('hibernate_sequence'), 'reserved', id FROM vocabulary WHERE name = 'Allele database status vocabulary';
+INSERT INTO vocabularyterm (id, name, vocabulary_id) SELECT nextval('hibernate_sequence'), 'autoloaded', id FROM vocabulary WHERE name = 'Allele database status vocabulary';
+INSERT INTO vocabularyterm (id, name, vocabulary_id) SELECT nextval('hibernate_sequence'), 'in_progress', id FROM vocabulary WHERE name = 'Allele database status vocabulary';
```

Comment:
Also looks like there are two missing FK's:

```
ALTER TABLE public.alleledatabasestatusslotannotation_aud
    ADD CONSTRAINT fko47ewqwil249eqgi82tcc3wxn FOREIGN KEY (id, rev) REFERENCES public.nameslotannotation_aud(id, rev);
```

and

```
ALTER TABLE public.alleledatabasestatusslotannotation
    ADD CONSTRAINT fkbjl9t9iwmxl7jgxuwpi8r3atc FOREIGN KEY (id) REFERENCES public.nameslotannotation(id);
```
   
Always need to test against what hibernate is going to do.

Reply:
Actually, it was the class definition that was wrong - it shouldn't be extending NameSlotAnnotation, just SlotAnnotation

Reply:
Was missing an FK constraint on the id field to the slotAnnotation table though - fixed now

Reply:
Ahh ok.